File: Day-01/python/0skxr/day01_2.py
numbers = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight","nine"]
value = 0
f = open("day1.txt", "r")
for line in f.readlines():
  current = []
  for i in range(len(line)):
    for n in numbers:
      if line[i:i + 5].startswith(n): current.append(str(numbers.index(n)))
    if line[i] in ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]: current.append(line[i])
  value += int(current[0] + current[-1])
print(value)

File: Day-01/java/Asedem/Day01_2.java
package de.asedem;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Main {

    private static final String[] numbers = {"one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};

    public static void main(String[] args) throws FileNotFoundException {
        int finalNumber = 0;
        try (Scanner scanner = new Scanner(new File("./inputs.txt"))) {
            while (scanner.hasNextLine()) {
                String text = scanner.nextLine();
                for (int i = 0; i < numbers.length; i++)
                    text = text.replaceAll(numbers[i], numbers[i] + (i + 1) + numbers[i]);
                text = text.replaceAll("\\D+","");
                finalNumber += (text.charAt(0) - '0') * 10 + text.charAt(text.length() - 1) - '0';
            }
        }
        System.out.println(finalNumber);
    }
}
